daffodil-core/src/main/scala/org/apache/daffodil/grammar/AlignedMixin.scala

```diff
@@ -138,7 +138,7 @@ trait AlignedMixin extends GrammarMixin { self: Term =>
   // The separator is text in some encoding, might not be the same as this term's encoding, and
   // the alignment will be left where that text leaves it.
   //
-  private lazy val priorAlignmentApprox: AlignmentMultipleOf = LV('priorAlignmentApprox) {
+  lazy val priorAlignmentApprox: AlignmentMultipleOf = LV('priorAlignmentApprox) {
```

Comment:
I have been trying Steve's suggestion to replace the elementChildren for-loop in `elementParseAndUnspecifiedLengthGenerateCode(g, cgState)` with a recursive call of `Runtime2CodeGenerator.generateCode(g.eGram, cgState)`.  I have confirmed that the padtest test case steps into an AlignmentFill with 32-bit alignment which makes Steve's suggestion better than calling isKnownToBeAligned (besides, I have already tried calling isKnownToBeAligned and found it is true for every element in the padtest test case).  

Unfortunately, the elementChildren for-loop in `elementParseAndUnspecifiedLengthGenerateCode(g, cgState)` was the key piece of logic used by the code generator to generate C code for elements.  I have been working on replacing the logic since yesterday, but it is taking a lot of time to get the C code generated correctly again.  The schema compiler and the grammar intermediate objects it produces really need to be refactored and given a walk API so that separable runtimes/back-ends can convert the intermediate objects into runtime objects (runtime1), generated code (runtime2), or other formal artifacts (see [DAFFODIL-2536](https://issues.apache.org/jira/browse/DAFFODIL-2536) and Refactoring for Separable Runtimes/Back-ends in [Roadmap for Upcoming Releases](https://cwiki.apache.org/confluence/display/DAFFODIL/Roadmap+for+Upcoming+Releases)).  For now, I'm trying to fix the logic to walk grammar objects manually class by class using a large match expression in Runtime2CodeGenerator.generateCode and further logic in generator trait methods as I've done before.

Reply:
I looked at padtest. The only term in there that would have isKnownToBeAligned false is this sequence:

<xs:sequence dfdl:alignment="4" dfdl:alignmentUnits="bytes"/>

That would have it false.  It would be true for all the elements.

I think that makes the point that a walker is needed so that a generator will encounter this object.